### DISCUSSION TITLE
Strong names for Orleans Binaries

### DISCUSSION THREAD

**Discussion by nickshrayer** [dotnet/orleans#7012] [Q&A]
Hello,
our application DLLs are all strongly-named. Thus they only can use other DLLs which have strong names. Unfortunately Orleans DLLs are not like those. How could I get Orleans binaries signed and with strong names?

**Comment by ReubenBond** [comment]
Orleans DLLs are signed by Microsoft, but they are not strong-named. There is a tool called StrongNamer which will add strong names to assemblies which do not yet have them. It may be suitable for your needs.

[Guidance states](https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/strong-naming) that we should not add a strong naming key to a library which is not already strong-named: 

> ❌ DO NOT add, remove, or change the strong naming key.

Given that, I believe we need other solutions, such as tooling to strong-name the published assemblies. Other options include rebuilding from source and including a strong-name key.
